core/src/test/java/io/confluent/rest/SaslTest.java

```diff
@@ -0,0 +1,262 @@
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLHandshakeException;
+import javax.security.auth.login.Configuration;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+
+import io.confluent.common.metrics.KafkaMetric;
+import io.confluent.rest.annotations.PerformanceMetric;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+public class SaslTest {
+  private static final Logger log = LoggerFactory.getLogger(SaslTest.class);
+
+  private File jaasFile;
+  private File loginPropertiesFile;
+  private String previousAuthConfig;
+  private SaslTestApplication app;
+  private CloseableHttpClient httpclient;
+  String httpUri = "http://localhost:8080";
+
+  @Before
+  public void setUp() throws Exception {
+    jaasFile = File.createTempFile("jaas", ".config");
+    loginPropertiesFile = File.createTempFile("login", ".properties");
+
+    String jaas = "c3 {\n"
+                  + "  org.eclipse.jetty.jaas.spi.PropertyFileLoginModule required\n"
+                  + "  debug=\"true\"\n"
+                  + "  file=\"" + loginPropertiesFile.getAbsolutePath() + "\";\n"
+                  + "};\n";
+    Files.write(
+        jaasFile.toPath(),
+        jaas.getBytes(),
+        StandardOpenOption.TRUNCATE_EXISTING
+    );
+
+    String loginProperties = "jay: kafka,Administrators\n"
+                             + "neha: akfak,Administrators\n"
+                             + "jun: kafka-\n";
+    Files.write(
+        loginPropertiesFile.toPath(),
+        loginProperties.getBytes(),
+        StandardOpenOption.TRUNCATE_EXISTING
+    );
+    previousAuthConfig = System.getProperty("java.security.auth.login.config");
+    Configuration.setConfiguration(null);
+    System.setProperty("java.security.auth.login.config", jaasFile.getAbsolutePath());
+    httpclient = HttpClients.createDefault();
+    TestMetricsReporter.reset();
+    Properties props = new Properties();
+    props.put(RestConfig.LISTENERS_CONFIG, httpUri);
+    props.put(RestConfig.METRICS_REPORTER_CLASSES_CONFIG, "io.confluent.rest.TestMetricsReporter");
+    configBasic(props);
+    TestRestConfig config = new TestRestConfig(props);
+    app = new SaslTestApplication(config);
+    app.start();
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    Configuration.setConfiguration(null);
+    if (previousAuthConfig != null) {
+      System.setProperty("java.security.auth.login.config", previousAuthConfig);
```

Comment:
In Kafka, we typically just clear the system property instead of restoring it. We do it that way because it's simpler and no test should rely on that property being set already. A minor point, so you can leave it if you prefer it that way.